TestEnvironment/demo-static-parent-issue/MyClass.php

```diff
@@ -0,0 +1,22 @@
+<?php
+class MyClass extends ParentClass
+{
+
+	public function load()
+	{
+		$dataProvider = DataProviderWithStaticFactory::getDataProvider();
+		$entity = $dataProvider->returnData();
+
+		$refreshedEntity = self::refreshEntity($entity);
+		$refreshedEntity->entityMethod(); // auto-complete does not work for this
```

Comment:
@pbyrne84 just to be sure - this PR is not another repro-case for previous issue, but rather an example of a different (but related) issue with static / parent calls